### PR TITLE
feat(api):  signin register

### PR DESCRIPTION
**Pull request by fsamin**
@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17160.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17163.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✘  
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17165.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17166.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17168.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17172.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

File: sdk/cdsclient/interface.go [hunk 1]
```diff
@@ -405,7 +405,7 @@ type Interface interface {
 	Requirements() ([]sdk.Requirement, error)
 	RepositoriesManagerInterface
 	ServiceClient
-	ServiceRegister(context.Context, sdk.Service) (*sdk.Service, error)
+	//ServiceRegister(context.Context, sdk.Service) (*sdk.Service, error)
```
Comment (yesnault) [inline sdk/cdsclient/interface.go:408]:
```suggestion
```

File: engine/api/services/dao.go [hunk 2]
```diff
@@ -160,10 +171,10 @@ func UpsertStatus(db gorpmapper.SqlExecutorWithTx, s sdk.Service, authSessionID
 		_, err := db.Exec(query, s.MonitoringStatus, s.ID)
 		return sdk.WithStack(err)
 	}
-	sessionID = &authSessionID
 	query := `INSERT INTO service_status(monitoring_status, service_id, auth_session_id) VALUES($1,$2, $3)
 	ON CONFLICT (service_id, auth_session_id) DO UPDATE SET monitoring_status = $1, service_id = $2, auth_session_id = $3`
-	_, err := db.Exec(query, s.MonitoringStatus, s.ID, sessionID)
+	log.Info(ctx, "updating service %d status for session %q", s.ID, authSessionID)
```
Comment (yesnault) [inline engine/api/services/dao.go:176]:
```suggestion
	log.Debug(ctx, "updating service %d status for session %q", s.ID, authSessionID)
```

**Review by yesnault — CHANGES_REQUESTED** [review]

**Review by yesnault — APPROVED** [review]

**Review by sguiheux — APPROVED** [review]

File: engine/api/services.go [hunk 3]
```diff
@@ -51,96 +54,88 @@ func (api *API) getServiceHandler() service.Handler {
...
+	log.Info(ctx, "Registering service %s(%d) %s, consumer: %s, session %s", srv.Name, srv.ID, strRegion, consumer.ID, session.ID)
 
-		// For hatchery service we need to check if there are workers that are not attached to an existing hatchery
-		// If some worker's parent consumer match current hatchery consumer we will attach this worker to the new hatchery.
-		if srv.Type == sdk.TypeHatchery {
-			if err := worker.ReAttachAllToHatchery(ctx, tx, *srv); err != nil {
-				return err
-			}
-		}
+	_, err = authentication.LoadSessionByID(ctx, tx, session.ID)
```
Comment (richardlt) [inline engine/api/services.go:115]: Why do we need this ? Maybe add a comment.

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17179.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

File: engine/service/types.go [hunk 4]
```diff
@@ -120,6 +120,7 @@ type Common struct {
 	CDNLogsURL           string
 	ServiceLogger        *logrus.Logger
 	GoRoutines           *sdk.GoRoutines
+	Region               string
```
[the remark this answers is not in the document]
Comment (fsamin) [reply engine/service/types.go:123]: yep in a next PR

File: engine/api/services.go (hunk 3 above)
Comment (fsamin) [reply engine/api/services.go:115]: it was a debugging step

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#17203.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✘

**Pull request merged**